core/src/test/java/nl/nn/adapterframework/doc/DocWriterNewIntegrationTest.java

```diff
@@ -2,37 +2,69 @@
 
 import java.io.BufferedWriter;
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.Writer;
 
+import javax.xml.XMLConstants;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.apache.logging.log4j.Logger;
 import org.junit.Ignore;
 import org.junit.Test;
+import org.xml.sax.InputSource;
 
 import nl.nn.adapterframework.doc.model.FrankDocModel;
 import nl.nn.adapterframework.doc.model.FrankElement;
 import nl.nn.adapterframework.doc.model.FrankElementStatistics;
+import nl.nn.adapterframework.doc.model.XsdVersion;
+import nl.nn.adapterframework.util.LogUtil;
 
-@Ignore("Test takes a long time to run, and gives little information")
 public class DocWriterNewIntegrationTest {
+	private static Logger log = LogUtil.getLogger(DocWriterNewIntegrationTest.class);
+
+	@Test
+	public void testStrict() throws Exception {
+		String schemaFileName = generateXsd(XsdVersion.STRICT);
+		validate(schemaFileName);
+	}
 
 	@Test
-	public void testXsd() throws IOException {
+	public void testCompatibility() throws Exception {
+		String schemaFileName = generateXsd(XsdVersion.COMPATIBILITY);
+		validate(schemaFileName);
+	}
+
+	String generateXsd(XsdVersion version) throws IOException {
 		FrankDocModel model = FrankDocModel.populate();
 		DocWriterNew docWriter = new DocWriterNew(model);
-		docWriter.init();
+		docWriter.init(version);
 		String xsdString = docWriter.getSchema();
-		File output = new File("testFrankdoc.xsd");
-		System.out.println("Output file of test xsd: " + output.getAbsolutePath());
+		File output = new File("FrankConfig-" + docWriter.getOutputFileName());
+		log.info("Output file of test xsd: " + output.getAbsolutePath());
 		Writer writer = new BufferedWriter(new FileWriter(output));
 		try {
 			writer.append(xsdString);
 		}
 		finally {
 			writer.close();
 		}
+		return output.getAbsolutePath();
+	}
+
+	private void validate(String schemaFileName) throws Exception {
```

Comment:
The validation package does validation against XML schema, but embeds this in a pipe. That code cannot be used easily in a unit test. Using plain SAX turns out not to be too difficult in the end.